### PR TITLE
i18n(fr): update `guides/typescript.mdx`

### PR DESCRIPTION
**Pull request by ArmandPhilippot**
#### Description (required)

Updates the French translation of `guides/typescript.mdx`:
* add changes from #9240, #10148 and #10197
* fix some inline codes (no space before colon)

#### Related issues & labels (optional)

- Suggested label: i18n

### CODE REVIEW

File: src/content/docs/fr/guides/typescript.mdx [hunk 1]
```diff
@@ -15,33 +15,44 @@ Vous n'avez pas besoin d'écrire du code TypeScript dans vos projets Astro pour
...
+```
+
+Vous pouvez créer le fichier `src/env.d.ts` comme convention pour ajouter des déclarations de types personnalisés, ou pour bénéficier des types Astro si vous n'avez pas de `tsconfig.json` :
+
+```ts title="src/env.d.ts"
+// Custom types declarations
+declare var myString: string;
 
-```typescript title="env.d.ts"
+// Astro types, not necessary if you already have a tsconfig.json
```
Comment (thomasbnt) [inline src/content/docs/fr/guides/typescript.mdx:48]:
```suggestion
// Types d'Astro, non nécessaire si vous avez déjà un tsconfig.json
```

**Review by thomasbnt — CHANGES_REQUESTED** [review]
Miss one translation, LGTM for the rest!

File: src/content/docs/fr/guides/typescript.mdx [hunk 2]
```diff
@@ -15,33 +15,44 @@ Vous n'avez pas besoin d'écrire du code TypeScript dans vos projets Astro pour
...
+  "extends": "astro/tsconfigs/base",
+  "include": [".astro/types.d.ts", "**/*"],
+  "exclude": ["dist"]
+}
+```
+
+Vous pouvez créer le fichier `src/env.d.ts` comme convention pour ajouter des déclarations de types personnalisés, ou pour bénéficier des types Astro si vous n'avez pas de `tsconfig.json` :
+
+```ts title="src/env.d.ts"
+// Custom types declarations
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/typescript.mdx:45]: This one too to be consistent:
```suggestion
Déclarations de types personnalisés
```

File: src/content/docs/fr/guides/typescript.mdx [hunk 3]
```diff
@@ -15,33 +15,44 @@ Vous n'avez pas besoin d'écrire du code TypeScript dans vos projets Astro pour
...
+  "extends": "astro/tsconfigs/base",
+  "include": [".astro/types.d.ts", "**/*"],
+  "exclude": ["dist"]
+}
+```
+
+Vous pouvez créer le fichier `src/env.d.ts` comme convention pour ajouter des déclarations de types personnalisés, ou pour bénéficier des types Astro si vous n'avez pas de `tsconfig.json` :
+
+```ts title="src/env.d.ts"
+Déclarations de types personnalisés
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/typescript.mdx:45]: Sorry, too quick... 😅 
```suggestion
// Déclarations de types personnalisés
```

**Review by thomasbnt — APPROVED** [review]
LGTM!

**Pull request merged**